content/soda-gettingstarted/soda-lab-setup/vagrant-setup.md

```diff
@@ -0,0 +1,249 @@
+flexibility of you and your team.
+
+### Requirements
+
+-   VirtualBox installed on machine
+
+
+## Gettiing Started
+
+Follow the below mentioned steps to get quickly started with vagrant enviroment
+
+
+- Install from the apt repository
+    
+    ``` bash
+        $ sudo apt install vagrant
+    ```
+
+### Install required plugins
+
+Required Vagrant plugins are:
+
+| Sno | Plugin Name                | Description                                                    |
+|-----|----------------------------|----------------------------------------------------------------|
+| 1   | vagrant-env                | Plugin to load environment variables from .env into ENV        |
+| 2   | vagrant-disksize           | Plugin to resize root disk                                     |
+| 3   | vagrant-persistent-storage | Creates a persistent storage and attaches it to guest machine. |
+
+
+-   Vagrant plugin to load environment variables from .env into ENV
+
+    ``` bash
+    $ vagrant plugin install vagrant-env
+    ```
+
+-   Vagrant disk size plugin to resize root disk
+
+    ``` bash
+        $ vagrant plugin install vagrant-disksize
+    ```
+
+    <div class="note">
+    <div class="title">
+
+    Note
+
+    </div>
+
+    *Example*
+
+    `config.disksize.size = '50GB'`
+
+    </div>
+
+    -   Vagrant plugin that creates a persistent storage and attaches it to
+        guest machine.
+
+    ``` bash
+    $ vagrant plugin install vagrant-persistent-storage
+    ```
+
+    <div class="note">
+
+    <div class="title">
+
+    <b>Note:</b>
+    </div>
+
+    *Example Usage*
+    <br>
+    <br>config.persistent_storage.enabled = true
+    <br>config.persistent_storage.location = "\~/development/sourcehdd.vdi"
+    <br>config.persistent_storage.size = 5000
+    <br>config.persistent_storage.mountname = 'mysql'
+    <br>config.persistent_storage.filesystem = 'ext4'
+    <br>config.persistent_storage.mountpoint = '/var/lib/mysql'
+    <br>config.persistent_storage.volgroupname = 'myvolgroup'
+
+    </div>
+
+    <div class="important">
+
+    <div class="title">
+
+    Important
+
+    </div>
+
+    After installing you can set the location and size of the persistent
+    storage.
+
+    The following options will create a persistent storage with 5000 MB,
+    named `docker`, mounted on `/var/lib/docker`, in a volume group called
+    `'myvolgroup'` [Persistent storage][]
+
+    ![image][]
+
+    </div>
+
+## Setup Demo Vagrant Project 
+
+
+-   Make a directory for vagrant using mkdir
+
+    ``` bash
+        $ mkdir vagrant
+        $ cd vagrant/
+    ```
+
+-   Make a directory inside Vagrant and give the name of VM & cd into it
+
+    ``` bash
+        $ mkdir ubuntu
+        $ cd ubuntu/
+    ```
+
+    <div class="important">
+
+    <div class="title">
+
+    Important
+
+    </div>
+
+    Discover [Vagrant Boxes][]
+
+    Search your required image
+
+    </div>
+
+-   Download the image of ubuntu-20.04 in directory
```

Comment:
SODA is not fully qualified for Ubuntu 20.04.
Maybe update the steps for Ubuntu 18.04/16.04 and associated changes.